### DISCUSSION TITLE
编译xquic是报错“implicit declaration of function ‘HMAC_CTX_new’”，当前openssl版本是1.0.2k-fips，需要升版本么？

### DISCUSSION THREAD

**Discussion by lonely-wind-nomiss** [alibaba/xquic#108] [Q&A] [answered]
[ 13%] Building C object CMakeFiles/xquic.dir/src/transport/xqc_cid.c.o
[ 14%] Building C object CMakeFiles/xquic.dir/src/transport/xqc_packet_parser.c.o
/root/gits/xquic/src/transport/xqc_packet_parser.c: In function ‘xqc_gen_reset_token’:
/root/gits/xquic/src/transport/xqc_packet_parser.c:1192:5: error: implicit declaration of function ‘HMAC_CTX_new’ [-Werror=implicit-function-declaration]
     HMAC_CTX *ctx = HMAC_CTX_new();
     ^
/root/gits/xquic/src/transport/xqc_packet_parser.c:1192:21: error: initialization makes pointer from integer without a cast [-Werror]
     HMAC_CTX *ctx = HMAC_CTX_new();
                     ^
/root/gits/xquic/src/transport/xqc_packet_parser.c:1193:5: error: implicit declaration of function ‘HMAC_CTX_reset’ [-Werror=implicit-function-declaration]
     HMAC_CTX_reset(ctx);
     ^
/root/gits/xquic/src/transport/xqc_packet_parser.c:1198:5: error: implicit declaration of function ‘HMAC_CTX_free’ [-Werror=implicit-function-declaration]
     HMAC_CTX_free(ctx);
     ^
cc1: all warnings being treated as errors

**Comment by lonely-wind-nomiss** [comment, marked as answer]
版本升级后依旧有问题
In function ‘xqc_ssl_enable_max_early_data’:
/root/gits/xquic/src/tls/xqc_tls.c:71:1: error: unknown type name ‘SSL_QUIC_METHOD’
 SSL_QUIC_METHOD xqc_ssl_quic_method;
 ^
/root/gits/xquic/src/tls/xqc_tls.c: In function ‘xqc_tls_init_server_ssl’:
/root/gits/xquic/src/tls/xqc_tls.c:249:5: error: implicit declaration of function ‘SSL_set_quic_early_data_context’ [-Werror=implicit-function-declaration]
     SSL_set_quic_early_data_context(ssl, (const uint8_t *)XQC_EARLY_DATA_CONTEXT, 
     ^
/root/gits/xquic/src/tls/xqc_tls.c: In function ‘xqc_tls_create_ssl’:
/root/gits/xquic/src/tls/xqc_tls.c:283:5: error: implicit declaration of function ‘SSL_set_quic_method’ [-Werror=implicit-function-declaration]
     ssl_ret = SSL_set_quic_method(ssl, &xqc_ssl_quic_method);
     ^
/root/gits/xquic/src/tls/xqc_tls.c:292:5: error: implicit declaration of function ‘SSL_set_quic_transport_params’ [-Werror=implicit-function-declaration]

**Reply by Kulsk** [reply]
it seems that a ssl library without quic is used. up to now, openssl do not support quic yet, check if babassl or boringssl is used.

**Reply by lhh-208** [reply]
解决了吗？

**Comment by lonely-wind-nomiss** [comment]
解决了，升级一下就好了，版本在1.1.0以上即可


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "alibaba/xquic"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年6月28日(星期二) 晚上8:28
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [alibaba/xquic] 编译xquic是报错“implicit declaration of function ‘HMAC_CTX_new’”，当前openssl版本是1.0.2k-fips，需要升版本么？ (Discussion #108)





 
解决了吗？
